draft-ietf-quic-tls.md

```diff
@@ -1115,6 +1118,74 @@ values in the following registries:
 
 --- back
 
+# Test Vectors for Initial Packet Encryption {#test-vectors-initial}
+
+This section shows sample packet encryption secrets so
+that implementations can be verified incrementally.
+
+{{initial-secrets}} contains the salt used for Initial Packet
+Encryption.
+Using the Initial Destination Connection ID 0x8394c8f03e515708, the
+derived 32 byte initial secret is:
+
+~~~
+  a5 72 b0 24 5a f1 ed df 5c 61 c6 e3 f7 f9 30 4c
+  a6 6b fb 4c aa f7 65 67 d5 cb 8d d1 dc 4e 82 0b
+~~~
+
+The labels generated by the HKDF-Expand-Label function are:
+
+~~~
+  quic client in: 00 20 0e 71 75 69 63 20 63 6c 69 65 6e 74 20
+                  69 6e
+
+  quic server in: 00 20 0e 71 75 69 63 20 73 65 72 76 65 72 20
+                  69 6e
+
+  quic key:       00 10 08 71 75 69 63 20 6b 65 79
+
+  quic iv:        00 0c 07 71 75 69 63 20 69 76
+
+  quic pn:        00 10 07 71 75 69 63 20 70 6e
+~~~
+
+The client initial secret is 32 bytes long:
+
+~~~
+  37 2c 95 89 52 fd 99 c5 76 85 b8 3c 72 ac 15 f0
+  80 a3 69 eb ec c6 8f 26 eb d7 61 4c b3 76 06 13
+~~~
+
+Using this secret, we get the following keys (16 bytes) and
+initialization vector (12 bytes):
+
+~~~
+  key:     e4 7c 2f 2c e2 c1 45 14 16 a9 34 90 ea fb 50 9c
+
+  pn key:  02 52 be 17 90 16 69 06 71 33 de 18 e6 39 4e 68
+
+  IV:      d9 9f 59 69 9e 3c 32 9a f1 95 06 52
+~~~
+
+The server initial secret is 32 bytes long:
+
+~~~
+  f2 41 f2 57 2e fe ee 53 ca 69 0c e5 b2 85 ab 34
+  e9 39 e8 fe 98 d7 14 6b b9 fc 78 e9 9a 1b 41 ab
+~~~
+
+Using this secret, we get the following keys (16 bytes) and
+initialization vector (12 bytes):
+
+~~~
+  key:     99 ff fd 2c 3e 4a cd ae 3d 32 02 1c 75 38 86 c0
+
+  pn key:  14 c2 70 2b 54 48 80 d3 9d ab d7 dd c8 eb cd 9a
+
+  IV:      27 92 24 e5 85 fd 84 e9 e0 70 ea 9a
+~~~
+
+
```

Comment:
Yes, having something like that in the spec would be useful. We need an independent verification before publishing these values.

Also, having the PN key is fine, but we probably need a complete packet example to ensure that the nonce is properly extracted and the PN properly decrypted.

Reply:
In fact, I think that it would be better to start with the test vector in EKR's message "PNE Test Vector" sent on the quic list on 7/18/2018.

Reply:
@huitema my objective was to have the PN key in this section and then use it for PNE but this PR was just for initial test vectors.  I agree we need independent verification before we publish.